projects/portal/src/app/pages/dialogs/delegate/undelegate-form-dialog/undelegate-form-dialog.component.ts

```diff
@@ -50,18 +51,22 @@ export class UndelegateFormDialogComponent implements OnInit {
     );
     this.unbondingDelegation$ = combineLatest([this.cosmosSDK.sdk$, address$]).pipe(
       mergeMap(([sdk, address]) => {
-        console.log(this.validator);
         const validatorAddress = this.validator?.operator_address;
-        const validatorAccAddress = cosmosclient.ValAddress.fromString(validatorAddress || '');
+        if (!validatorAddress) {
+          return of(undefined)
+        }
+        const validatorAccAddress = cosmosclient.ValAddress.fromString(validatorAddress);
         const unbondingDelegation = rest.staking.unbondingDelegation(
           sdk.rest,
           validatorAccAddress,
           address,
         );
-        console.log({ address, validatorAccAddress, unbondingDelegation });
-        return unbondingDelegation;
+        return unbondingDelegation
+      }),
```

Comment:
This is a matter of taste, but I think it would be less confusing to have the string in valAddress and the variable name of the type in the @cosmos-client/core library as follows.

validatorAddress -> validatorAddressString or valAddreessString
validatorAccAddress -> validatorAddress or valAddress

In particular, I feel validatorAccAddress would be very confusing.